### PR TITLE
Team 3

### PR DESCRIPTION
**Pull request by Zorni3001**

### CODE REVIEW

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 1]
```diff
@@ -4,28 +4,38 @@
...
 		return builder.toString();
 	}
 
-	private char[] make(int i, int j) {
-		int amount = ((i - j) / 2);
-		char[] chars = new char[amount + j + 1];
-		if (amount > 0) {
-			Arrays.fill(chars, 0, amount, ' ');
+	private char[] generateLine(int maxWidth, int currentWidth) {
+		int center = ((maxWidth - currentWidth) / 2);
```
Comment (fabianSorn) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:26]: We think those are  the amount of whitespaces in the front, not the center of the line

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 2]
```diff
@@ -4,28 +4,38 @@
...
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
+	private final char WHITESPACE = ' ';
+
+	private final char STAR = '*';
+
+	private final Character NEW_LINE = '\n';
+
+	public String print(int width) {
+		if (invalidInput(width)) return null;
```
Comment (fabianSorn) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:14]: Inconsistent brackets compared to single line ifs in line 30

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 3]
```diff
@@ -4,28 +4,38 @@
...
-		StringBuilder builder = new StringBuilder(new String(make(n, n)));
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
+	private final char WHITESPACE = ' ';
+
+	private final char STAR = '*';
+
+	private final Character NEW_LINE = '\n';
+
+	public String print(int width) {
```
Comment (fabianSorn) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:13]: bg,f8f8f8-flat,750x,075,f-pad,75x100,f8f8f8 u7

this doesnt print but just builds the string

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 4]
```diff
@@ -4,28 +4,38 @@
...
-		Arrays.fill(chars, amount, amount + j, '*');
-		chars[chars.length - 1] = '\n';
-		return chars;
+		Arrays.fill(line, center, center + currentWidth, STAR);
+		line[line.length - 1] = NEW_LINE;
+
+		return line;
+	}
+
+	private Boolean invalidInput(int width){
```
Comment (fabianSorn) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:37]: Why not the primitive type `boolean`, maybe `isInvalidWidth` might have been better

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java (hunk 3 above)
Comment (DragonflyCode11) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:13]: This method does not print. It just creates a String

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 5]
```diff
@@ -4,28 +4,38 @@
...
 public class Diamond {
 
-	public String print(int n) {
-		if (n <= 0 || n % 2 == 0) {
-			return null;
-		}
-		StringBuilder builder = new StringBuilder(new String(make(n, n)));
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
+	private final char WHITESPACE = ' ';
```
Comment (Bananamafia) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:7]: Konstanten benutzt 👍

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 6]
```diff
@@ -4,28 +4,38 @@
...
+
+	private final char STAR = '*';
+
+	private final Character NEW_LINE = '\n';
```
Comment (Bananamafia) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:11]: char statt Character

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 7]
```diff
@@ -4,28 +4,38 @@
...
+
+	public String print(int width) {
+		if (invalidInput(width)) return null;
+		
+		StringBuilder builder = new StringBuilder(new String(generateLine(width, width)));
+		for (int currentHeight = width - 2; currentHeight > 0; currentHeight -= 2) {
+			char[] chars = generateLine(width, currentHeight);
```
Comment (DragonflyCode11) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:18]: chars is not a good name. Rename to line.

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java (hunk 2 above)
Comment (Bananamafia) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:14]: Fehlerhandling 👍

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 8]
```diff
@@ -4,28 +4,38 @@
...
+
+	private final char STAR = '*';
+
+	private final Character NEW_LINE = '\n';
+
+	public String print(int width) {
+		if (invalidInput(width)) return null;
+		
+		StringBuilder builder = new StringBuilder(new String(generateLine(width, width)));
+		for (int currentHeight = width - 2; currentHeight > 0; currentHeight -= 2) {
```
Comment (fabianSorn) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:17]: `currentHeight` is not really fitting since it represents the width

Comment (Bananamafia) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:17]: currentWidth, wenn weiter unten auch schon currentWidth

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java (hunk 1 above)
Comment (DragonflyCode11) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:26]: This is not the center. It is just the amount of whitespaces

**Pull request merged**